### PR TITLE
Move card loading logic into hui-card

### LINKED ISSUE #21009: Visibility option only functional on top level card, not on nested card
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

as title, the new (2024.6) visibility option for cards does not work when doing below snippet

the card with the visibility option set is visible always.

using that exact configuration as separate cards works as expected

### Describe the behavior you expected

have the visibility option work on nested card

### Steps to reproduce the issue

1. see yaml
2.
3.
...

### What version of Home Assistant Core has the issue?

2024.6.0

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

any

### Which operating system are you using to run this browser?

sonoma 14.5

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

```yaml
    - type: vertical-stack
      cards:
        - type: entities
          entities:
            - input_boolean.test_2

        - type: conditional
          conditions:
            - condition: state
              entity: input_boolean.test_2
              state: 'on'
          card:
            type: entities
            title: Type Conditional
            entities:
              - light.zitkamer

        - type: entities
          title: Visibility
          visibility:
            - condition: state
              entity: input_boolean.test_2
              state: 'off'
          entities:
            - light.zitkamer
```

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

- [x] Move all the rebuild card logic to `hui-card`
- [x] Use `hui-card` in vertical-stack, horizontal-stack, grid-stack
- [x] Use `hui-card` in conditional-card
- [x] Use `hui-card` in card dialog
- [x] Use `hui-card` in todo list
- [x] Use `hui-card` in entity-filter-card
- [x] Use `hui-card` in gallery

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #21009
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **New Features**
  - Replaced MWC buttons with HA buttons in demo cards.
  - Introduced `editMode` property to various card components for better editing control.

- **Bug Fixes**
  - Improved event handling for demo configuration updates.
  - Enhanced error handling and visibility control in card components.

- **Refactor**
  - Updated imports and type declarations to use `HuiCard` instead of `LovelaceCard`.
  - Simplified card creation and configuration logic across multiple components.

- **Chores**
  - Removed duplicate and unnecessary imports for better code maintainability.

### CODE REVIEW

File: src/panels/lovelace/editor/card-editor/hui-dialog-delete-card.ts [hunk 1]
```diff
@@ -45,10 +45,11 @@ export class HuiDialogDeleteCard extends LitElement {
           ${this._cardConfig
             ? html`
                 <div class="element-preview">
-                  <hui-card-preview
+                  <hui-card
                     .hass=${this.hass}
                     .config=${this._cardConfig}
-                  ></hui-card-preview>
+                    .editMode=${true}
```
Comment (bramkragten) [inline src/panels/lovelace/editor/card-editor/hui-dialog-delete-card.ts:51]:
```suggestion
                    editMode
```

**Comment by bramkragten** [comment]
Can we also use it on the device page?

**Comment by piitaya** [comment]
> Can we also use it on the device page?

We are using `ha-device-entities-card` which are using entity rows. It's not the entities card.

File: src/panels/lovelace/cards/hui-entity-filter-card.ts [hunk 2]
```diff
@@ -173,10 +173,10 @@ export class HuiEntityFilterCard
...
 
     if (!this.lastChild) {
-      this._element.setConfig({
+      this._element.config = {
         ...this._baseCardConfig!,
         entities: entitiesList,
-      });
+      };
       this._oldEntities = entitiesList;
     } else if (this._element.tagName !== "HUI-ERROR-CARD") {
```
[the remark this answers is not in the document]
Comment (piitaya) [reply src/panels/lovelace/cards/hui-entity-filter-card.ts:181]: Right! I need to figure out what this code for to see how to refactor it.

File: src/panels/lovelace/editor/card-editor/hui-dialog-delete-card.ts (hunk 1 above)
Comment (piitaya) [reply src/panels/lovelace/editor/card-editor/hui-dialog-delete-card.ts:51]: edit mode isn't a attribute for now but I will convert it.

**Comment by karwosts** [comment]
Sounds like this might fix #17266 but I'm not certain, will just drop this link here in case you wanted to collect issues related to card loading.

**Comment by piitaya** [comment]
@karwosts Yes, I hope it will fix all related issues with ll-rebuild event as it's handle by `hui-card` now.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**